astropy/coordinates/distances.py

```diff
@@ -153,37 +152,33 @@ def __new__(
                 # the log of the distance
                 meanlogval = np.log10(value.value).mean()
                 if meanlogval > 6:
-                    unit = u.Mpc
+                    value <<= u.Mpc
```

Comment:
I'm a bit worried about the patch coverage claiming this is not covered - I thought in general that was more reliable.

Reply:
Hmm. Weird. The test doesn't use remote data or optional dep, so should have been covered. Is coveragerc setting accidentally ignoring this module or something?

Reply:
Nope, says it is covered. 🤷 

https://app.codecov.io/gh/astropy/astropy/pull/15900/blob/astropy/coordinates/distances.py